src/pages/heeyeolKim/Login/Login.js

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+
+import { withRouter } from 'react-router-dom';
+//import { Link } from 'react-router-dom';
+
+import '../Login/Login.scss';
+
+import appStore1 from '../../../images/heeyeolKim/Login/appStore1.png';
+import googlePlay2 from '../../../images/heeyeolKim/Login/googlePlay2.png';
+import loginPageImage from '../../../images/heeyeolKim/Login/loginPageImage.png';
+import logoImage from '../../../images/heeyeolKim/Login/logoImage.png'
+
+class Login extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            idValue: "",
+            pwValue: "",
+            bottonOpacity: "", // 버튼 활성화를 위한 state
+            buttonOn: false, // 메인으로 가기 위한 state. input에 요건이 충족하면 opacity가 1이 되고, buttonOn이 true가 되어 메인으로 갈 수 있다.
+        }
+    }
+
+    // handleActivation = () => {
+    //     //const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+    //     if (this.state.idValue.includes("@") && this.state.pwValue.length >= 5) {
+    //         this.setState({
+    //             buttonOn: true,
+    //             buttonOpacity: 1,
+    //         })
+    //     } else {
+    //         this.setState({
+    //             buttonOn: false,
+    //             buttonOpacity: 0.3,
+    //         })
+    //     }
+    // }
+
+    handleActivation = () => {
+        //const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+        ( this.state.idValue.includes("@") && this.state.pwValue.length >= 5 )
+        ? this.setState({
+            buttonToGo: true,
+            buttonOpacity: 1,
+        })
+        : this.setState({
+            buttonToGo: false,
+            buttonOpacity: 0.3,
+        })
+    }
```

Comment:
```suggestion
    handleActivation = () => {
        const { idValue, pwValue } = this.state;
        const isLoginValid =  idValue.includes("@") && pwValue.length >= 5;
        
        this.setState({
          bunttonToGo : isLoginValid
        })
    }
```
- 제 코드 어때요?
- 조건이 길어지는 경우 변수를 활용해서 좀 더 예쁘게 작성해볼 수 있습니다.
- boolean 데이터 효과적으로 활용해보기!
- opacity 는 위 리뷰 참고해서 수정해보시기 바랍니다.

Reply:
반영 했습니다 😊